src/hats_import/catalog/run_import.py

```diff
@@ -148,3 +154,23 @@ def run(args, client):
             step_progress.update(1)
             assert is_valid_catalog(args.catalog_path)
             step_progress.update(1)
+
+
+def create_data_thumbnail(args, client, resume_plan):
+    """Create a thumbnail of the data in the catalog, top row from each partition."""
+    output_catalog_path = get_upath(args.catalog_path)
+    dataset_subdir = output_catalog_path / "dataset"
+    _, dataset = io.read_parquet_dataset(
+        dataset_subdir,
+        ignore_prefixes=["intermediate", "_common_metadata", "_metadata"],
+        exclude_invalid_files=True,
+    )
+    futures = [
+        client.submit(lambda pq_path: pd.read_parquet(pq_path, engine="pyarrow").head(1), p)
+        for p in dataset.files
+    ]
+    resume_plan.wait_for_thumbnail(futures)
+    if futures:
+        thumbnail = pd.concat(client.gather(futures), ignore_index=True)
```

Comment:
If you're building this with pandas, then it won't have the same schema as the leaf parquet files. 

- Things are written out with pyarrow parquet, so they will NOT have pandas metadata, but WILL have whatever additional metadata has been provided in the `use_schema_file` contents.
- This uses flat pandas, and so will clobber any nesting that had occurred, if importing from nested parquet files.

Reply:
I see.  I'm using Pandas here because it's what I use most often, but it's also true that this PR introduces Pandas where it didn't exist before.  I see that Pandas isn't formally in the `requirements.txt` either.  If this needs to be rewritten without Pandas, I can do it, using `pq.read_table` and `pa.concat_tables`, I suppose.